src/EnergyPlus/CondenserLoopTowers.cc

```diff
@@ -5827,8 +5828,16 @@ namespace CondenserLoopTowers {
             // calculate residual based on a balance where Twb + Ta + Tr = Node(WaterInletNode)%Temp
             return (Twb + Tapproach + Trange) - state.dataLoopNodes->Node(this->WaterInletNodeNum).Temp;
         };
+
+        // select bounds so range calculation through regula falsi can be determined
+        for (int upperTrIncr = 0; upperTrIncr <= 100; ++upperTrIncr) {
+            if (f(lowerTr) * f(upperTr + upperTrIncr) < 0) {
+                upperTr += upperTrIncr;
+                break;
+            }
+        }
```

Comment:
1) f(lowerTr) is a constant, could calculate that once before this loop
2) how high can upperTr go in this loop? 122 C? It would break sooner than that but it seems that an upper limit on Tr could be used instead? If this function is smooth, RootSolver will find the solution rather quickly, even if upperTr is large. Have you tried increasing VSTowerMaxRangeTemp to something higher than 22.2222? Is a number higher than that OK with respect to this cooling tower model? Yes it seems to work, but ask the harder questions now.

Reply:
Thanks for the quick review!
1. I pushed a commit to address that
2. I tried 30 and 50, both seemed better but there was still a few instances where `SolFla == -2`. Also, this kind of worked for this model and these coefficient, it might not work for another model or other set of coefficient (user-defined YorkCalc model), so I thought that perhaps having a "variable" upper bound would solve this issue.